src/main.py

```diff
@@ -21,28 +25,85 @@
 
 from src.api.aws_translate_api import AWSTranslateAPI
 
+# Function to log progress
+class TqdmToLogger:
+    def __init__(self, logger, level=logging.INFO):
+        self.logger = logger
+        self.level = level
+        self.last_message = None
+
+    def write(self, message):
+        message = message.strip()
+        if message != self.last_message:
+            self.last_message = message
+            self.logger.log(self.level, message)
+
+    def flush(self):
+        pass
+
+def get_env_var(name):
+    value = os.environ.get(name)
+    if not value:
+        logging.error(f"Error: Environment variable '{name}' is required but not set.")
+        sys.exit(1)
+    return value
+    
 def run():
+    logging.basicConfig(
+        level=logging.INFO,
+        handlers=[
+            logging.FileHandler("process.log"),
+            logging.StreamHandler(sys.stdout)
+        ]
+    )
+    logger = logging.getLogger()
```

Comment:
The main class should not be creating the logger. This should be defined behind an interface.

Reply:
TODO